classes/log/Repository.php

```diff
@@ -0,0 +1,187 @@
+<?php
+/**
+ * @file classes/log/Repository.php
+ *
+ * Copyright (c) 2014-2021 Simon Fraser University
+ * Copyright (c) 2000-2021 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class Repository
+ *
+ * @ingroup log
+ *
+ * @brief Operations for retrieving and modifying EmailLogEntry objects.
+ */
+
+namespace PKP\log;
+
+use APP\core\Application;
+use APP\facades\Repo;
+use APP\submission\Submission;
+use Illuminate\Support\Facades\Mail;
+use PKP\core\Core;
+use PKP\facades\Locale;
+use PKP\log\core\EmailLogEventType;
+use PKP\mail\Mailable;
+use PKP\user\User;
+use Illuminate\Support\Facades\DB;
+
+class Repository
+{
+    private EmailLogEntry $model;
+
+    public function __construct(EmailLogEntry $model)
```

Comment:
(feel free to use constructor promotion if you like)

Reply:
updated